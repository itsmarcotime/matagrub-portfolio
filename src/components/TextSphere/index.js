import React, { useEffect }from "react";
import TagCloud from "TagCloud";

const TextSphere = () => {
    useEffect(() => {
        return () => {
            const container = ".tagcloud";
            const texts = [
                "HTML",
                "CSS",
                "JavaScript",
                "MongoDB",
                "Express.js",
                "React.js",
                "Node.js",
                "MySQL",
                "Sequelize",
                "Jquery",
                "ES6",
                "Python",
                "GIT",
                "GITHUB",
            ];

            const options = {
                radius: 300,
                maxSpeed: "normal",
                initSpeed: "normal",
                keep: true,
            };

            TagCloud(container, texts, options);
        };
    }, []);

    return (
        <div className="text-sphere">
            <span className="tagcloud"></span>
        </div>
    );
};

export default TextSphere;