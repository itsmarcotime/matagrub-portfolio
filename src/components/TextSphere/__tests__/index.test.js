import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TextSphere from "..";

afterEach(cleanup);

describe('about component', () => {
    //baseline render
    it('renders', () => {
        render(<TextSphere />);
    });

    //snapshot test
    it('matches snapshot on DOM', () => {
        const { asFragment } = render(<TextSphere />);

        expect(asFragment()).toMatchSnapshot();
    });
});