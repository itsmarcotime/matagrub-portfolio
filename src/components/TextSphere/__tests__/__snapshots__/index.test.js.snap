// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about component matches snapshot on DOM 1`] = `
<DocumentFragment>
  <div
    class="text-sphere"
  >
    <span
      class="tagcloud"
    />
  </div>
</DocumentFragment>
`;
